modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/raft/MetaStorageLearnerListener.java

```diff
@@ -56,6 +58,11 @@ public void onWrite(Iterator<CommandClosure<WriteCommand>> iter) {
         }
     }
 
+    @Override
+    public void onBeforeApply(Command command) {
+        writeHandler.beforeApply(command);
```

Comment:
Why before and not on command execution?

Reply:
Because we need to change the command before it's saved to log